packages/hardhat-ethers/src/helpers.ts

```diff
@@ -248,6 +244,7 @@ export async function getContractFactoryByAbiAndBytecode(
     abi
   );
 
+  const { ContractFactory } = require("ethers") as typeof ethers;
```

Comment:
I think the point of dynamic requires is just to make Hardhat's startup faster. IMO moving this `require` this way is a micro-optimization that hurts readability and consistency. It's not _bad_, but it feels like an unnecessary change.